webpack/assets/javascripts/react_app/components/hosts/vms/storage/Disk.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import VMStorageActions from '../../../../actions/VMStorageActions';
+import { Button } from 'react-bootstrap';
+import Select from '../../../common/forms/Select';
+import Checkbox from '../../../common/forms/Checkbox';
+import TextInput from '../../../common/forms/TextInput';
+
+class Disk extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  removeDisk(event) {
+    VMStorageActions.removeDisk(this.props.controllerId, this.props.id);
+  }
+
+  selectableStores() {
+    return this.props.datastores.map((attribute) => {
+      const key = Object.keys(attribute)[0];
+      const value = Object.values(attribute)[0];
+
+      return (<option key={key} value={key}>{value}</option>);
+    });
+  }
+
+  selectablePods() {
+    return this.props.storagePods.map((attribute) => {
+      const key = Object.keys(attribute)[0];
+      const value = Object.values(attribute)[0];
+
+      return (<option key={key} value={key}>{value}</option>);
+    });
+  }
+
+  updateDiskAttribute(attribute, event) {
+    const value = event.target.value;
+    let attributes = Object.assign({}, this.props);
+
+    attributes[attribute] = value;
+    VMStorageActions.updateDisk(this.props.controllerId, this.props.id, attributes);
+  }
+
+  updateDiskBooleanAttributes(attribute, event) {
+    let attributes = Object.assign({}, this.props);
+
+    attributes[attribute] = !this.props[attribute];
+    VMStorageActions.updateDisk(this.props.controllerId, this.props.id, attributes);
```

Comment:
do you really need all props? I would imagine you could be able to update one attribute at a time.